include/zephyr/rtio/rtio.h

```diff
@@ -1444,7 +1440,7 @@ static inline int z_impl_rtio_submit(struct rtio *r, uint32_t wait_count)
 			 "semaphore was reset or timed out while waiting on completions!");
 	}
 #else
-	while (rtio_cqe_consumable(r) < wait_count) {
+	while (atomic_get(&r->cq_count) < cq_count) {
 		Z_SPIN_DELAY(10);
 		k_yield();
```

Comment:
Not related to this PR, but didn't we have an issue here in native_posix since we don't sleep and time doesn't move forward? I thought we fixed it.

Reply:
Z_SPIN_DELAY does just that